pkg/storage/allocator_scorer.go

```diff
@@ -162,7 +164,7 @@ func (c candidate) compare(o candidate) float64 {
 		}
 		return -4
 	}
-	if c.diversityScore != o.diversityScore {
+	if math.Abs(c.diversityScore-o.diversityScore) > epsilon {
```

Comment:
I think you should create a helper function like `compareDiversityScores` to encapsulate this and provide more clarity to those reading the code.

Reply:
This isn't the only case where we compare doubles. It isn't feasible to have helper function for double field. I prefer a system wide helper function (as I've done on multiple occasions in the past) but wasn't sure where to put it.

Reply:
I'm not sure either, but regardless, I'd just add a tiny function at bottom of this file:
```
func scoresAlmostEqual(val1, val2 double) bool {
  return math.Abs(val1-val2) <= epsilon
}
```
Trying to create some general, system-wide function is over-kill IMO, especially since you want this function to use a particular epsilon appropriate for this particular case.

Reply:
OK, done.